dockers/ipu-pipelines-runner/installdependencies.sh

```diff
@@ -0,0 +1,282 @@
+#!/bin/bash
+
+# Modified from https://github.com/microsoft/azure-pipelines-agent/blob/d2acd5f77c6b3914cdb6ed0e5fbea672929c7da9/src/Misc/layoutbin/installdependencies.sh
+
+user_id=`id -u`
+
+if [ $user_id -ne 0 ]; then
+    echo "Need to run with sudo privilege"
+    exit 1
+fi
+
+# Determine OS type
+# Debian based OS (Debian, Ubuntu, Linux Mint) has /etc/debian_version
+# Fedora based OS (Fedora, Redhat, Centos, Oracle Linux 7) has /etc/redhat-release
+# SUSE based OS (OpenSUSE, SUSE Enterprise) has ID_LIKE=suse in /etc/os-release
+
+function print_errormessage()
+{
+    echo "Can't install dotnet core dependencies."
+    echo "You can manually install all required dependencies based on following documentation"
+    echo "https://docs.microsoft.com/en-us/dotnet/core/dependencies?pivots=os-linux&tabs=netcore31"
+}
+
+function print_rhel6message()
+{
+    echo "We did our best effort to install dotnet core dependencies"
+    echo "However, there are some dependencies which require manual installation"
+    echo "You can install all remaining required dependencies based on the following documentation"
+    echo "https://github.com/dotnet/core/blob/master/Documentation/build-and-install-rhel6-prerequisites.md"
+}
+
+function print_rhel6errormessage()
+{
+    echo "We couldn't install dotnet core dependencies"
+    echo "You can manually install all required dependencies based on following documentation"
+    echo "https://docs.microsoft.com/en-us/dotnet/core/dependencies?pivots=os-linux&tabs=netcore31"
+    echo "In addition, there are some dependencies which require manual installation. Please follow this documentation"
+    echo "https://github.com/dotnet/core/blob/master/Documentation/build-and-install-rhel6-prerequisites.md"
+}
+
+if [ -e /etc/os-release ]
+then
+    echo "--------OS Information--------"
+    cat /etc/os-release
+    echo "------------------------------"
+
+    if [ -e /etc/debian_version ]
+    then
+        echo "The current OS is Debian based"
+        echo "--------Debian Version--------"
+        cat /etc/debian_version
+        echo "------------------------------"
+
+        # prefer apt over apt-get
+        command -v apt
+        if [ $? -eq 0 ]
+        then
+            apt update && apt install -y liblttng-ust0 libkrb5-3 zlib1g
+            if [ $? -ne 0 ]
+            then
+                echo "'apt' failed with exit code '$?'"
+                print_errormessage
+                exit 1
+            fi
+
+	        # debian 10 uses libssl1.1
+            # debian 9 uses libssl1.0.2
+            # other debian linux use libssl1.0.0
```

Comment:
us is still valid if you use Ubuntu as the base?

Reply:
This script is directly pulled from https://github.com/microsoft/azure-pipelines-agent/blob/d2acd5f77c6b3914cdb6ed0e5fbea672929c7da9/src/Misc/layoutbin/installdependencies.sh, I can see if I can remove the modifications I made and download a copy during container build instead of committing it here?

Reply:
If it is really just a copy, let download it on the fly and reduce mount of code we need maintain...